src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1381,6 +1387,13 @@ else if (ppolicy == ProjectionPolicy.NONE && nativeCRS != null) {
                 schema = typeBuilder.buildFeatureType();
             }
 
+            // applying wrappers using implementations of RetypeFeatureTypeCallback
+            for (RetypeFeatureTypeCallback callback :
+                    GeoServerExtensions.extensions(RetypeFeatureTypeCallback.class)) {
+                if (SimpleFeatureSource.class.isAssignableFrom(fs.getClass()))
```

Comment:
Same advise on try / catch around to catch and report errors when calling any extension